packages/base-contract/src/index.ts

```diff
@@ -150,6 +159,38 @@ export class BaseContract {
         }
         return rawEncoded;
     }
+    public async evmExecAsync(input: Buffer): Promise<string> {
```

Comment:
This looks great! One thought that I have is that it might occasionally be useful to specify the bytecode that is being used. For example, some of our tests will test that the `codesize` of the contract is within the limits of what the evm will allow (`codesize < 24576`). In such a case, we could memoize the bytecode that is being used in the contract and avoid using `getContractCodeAsync`.

Reply:
Yeah, it would be ideal if we could pass bytecode at instantiation. I tried this earlier, but found that it created too many disruptive changes. Also, the deployed bytecode and deploy*able* bytecode is different; we currently store only deploy**able** bytecode in `@0x/contract-artifacts`, and we need deploy**ed** bytecode for running calls on contract functions, so we would have to store both. @dekz  and @fabioberger felt strongly that would bloat the `@0x/contract-artifacts` package too much, so we opted for always calling `getContractCodeAsync` instead.